### PR TITLE
RavenDB-22614 Add azure queue storage connection string validation and syntax helper

### PR DESCRIPTION
**Pull request by kalczur**
### Issue link

https://issues.hibernatingrhinos.com/issue/RavenDB-22614/Connection-strings-view-becomes-unavailable-for-invalid-AQS-ETL-conn.-str.-Misleading-success-messages

### Type of change

- [x] Bug fix
- [ ] Regression bug fix
- [ ] Optimization
- [ ] New feature

### How risky is the change?

- [x] Low 
- [ ] Moderate 
- [ ] High
- [ ] Not relevant

### Backward compatibility

- [x] Non breaking change
- [ ] Ensured. Please explain how has it been implemented?
- [ ] Breaking change
- [ ] Not relevant

### Is it platform specific issue?

- [ ] Yes. Please list the affected platforms.
- [x] No

### Documentation update

- [ ] This change requires a documentation update. Please mark the issue on YouTrack using `Documentation Required` tag.
- [x] No documentation update is needed 

### Testing by Contributor

- [ ] Tests have been added that prove the fix is effective or that the feature works
- [ ] Internal classes added to the test class (e.g. entity or index definition classes) have the lowest possible access modifier (preferable `private`) 
- [x] It has been verified by manual testing

### Testing by RavenDB QA team

- [ ] This change requires a special QA testing due to possible performance or resources usage implications (CPU, memory, IO). Please mark the issue on YouTrack using `QA Required` tag.
- [x] No special testing by RavenDB QA team is needed

### Is there any existing behavior change of other features due to this change?

- [ ] Yes. Please list the affected features/subsystems and provide appropriate explanation
- [x] No

### UI work

- [ ] It requires further work in the Studio. Please mark the issue on YouTrack using `Studio Required` tag.
- [x] No UI work is needed

### CODE REVIEW

File: src/Raven.Studio/typescript/components/pages/database/settings/connectionStrings/editForms/AzureQueueStorageConnectionString.tsx [hunk 1]
```diff
@@ -253,7 +267,27 @@ const schema = yupObjectSchema<FormData>({
...
+                .string()
+                .nullable()
+                .when("$authType", {
+                    is: "connectionString",
+                    then: (schema) =>
+                        schema
+                            .required()
+                            .test(
+                                "connectionString",
+                                "Please define all required fields: DefaultEndpointsProtocol, AccountName, AccountKey and QueueEndpoint",
```
Comment (Danielle9897) [inline src/Raven.Studio/typescript/components/pages/database/settings/connectionStrings/editForms/AzureQueueStorageConnectionString.tsx]: If I understand correctly from the code [here](https://github.com/ravendb/ravendb/blob/6ae5d69e9888ff917db0e1167544ca0abf549013/src/Raven.Client/Documents/Operations/ETL/Queue/AzureQueueStorageConnectionSettings.cs#L65)
then, as I wrote in the [documentation PR](https://github.com/ravendb/docs/blob/9dfdd8399fd50cfd257da7aa51b5bddad612df23/Documentation/6.1/Raven.Documentation.Pages/server/ongoing-tasks/etl/queue-etl/azure-queue.markdown?plain=1#L60),  only these are mandatory:
       * `AccountName`
       * `DefaultEndpointsProtocol`
       * `QueueEndpoint` (when using HTTP protocol)

isn't that the case ?

Comment (kalczur) [reply src/Raven.Studio/typescript/components/pages/database/settings/connectionStrings/editForms/AzureQueueStorageConnectionString.tsx]: If AccountKey is not present, the QueueServiceClient constructor throws an error. I checked it by testing it from the Connections string view in Stuido.

https://github.com/ravendb/ravendb/blob/8d92390c05d5841b64dae29202b375a43acbae73/src/Raven.Server/Documents/ETL/Providers/Queue/QueueBrokerConnectionHelper.cs#L130-L131

https://learn.microsoft.com/en-us/azure/storage/common/storage-configure-connection-string#connect-to-the-emulator-account-using-the-shortcut

Comment (kalczur) [reply src/Raven.Studio/typescript/components/pages/database/settings/connectionStrings/editForms/AzureQueueStorageConnectionString.tsx]: But you're propably right about QueueEndpoint, I will check that.

Comment (kalczur) [reply src/Raven.Studio/typescript/components/pages/database/settings/connectionStrings/editForms/AzureQueueStorageConnectionString.tsx]: I removed QueueEndpoint from required fields

Comment (Danielle9897) [reply src/Raven.Studio/typescript/components/pages/database/settings/connectionStrings/editForms/AzureQueueStorageConnectionString.tsx]: @kalczur 
thanks, so I will add `AccountKey` to the mandatory params in the Docs PR

**Comment by kalczur** [comment]
Let's wait for the response to: https://github.com/ravendb/docs/pull/1880#issuecomment-2249624794

**Review by ml054 — APPROVED** [review]

**Pull request merged**